tools/prowgen/cmd/prowgen/main.go

```diff
@@ -133,17 +125,34 @@ func main() {
 					name = name[:len(name)-len(ext)] + "-" + flag.Arg(1) + ext
 
 					dst := filepath.Join(*inputDir, name)
-					if err := pkg.WriteJobConfig(&jobs, dst); err != nil {
-						exit(err, "writing branched config failed")
+					bytes, err := yaml.Marshal(jobs)
+					if err != nil {
+						log.Fatalf("Error marshaling jobs config: %v", err)
+					}
+
+					// Writes the job yaml
+					if err := ioutil.WriteFile(dst, bytes, 0o644); err != nil {
+						log.Fatalf("Error writing branches config: %v", err)
 					}
 				}
 			}
 
 			return nil
 		}); err != nil {
-			exit(err, "walking through the meta config files failed")
+			log.Fatalf("Walking through the meta config files failed: %v", err)
 		}
 	} else {
+		// Set INPUT and OUTPUT env vars for the pre-process and post-process
+		// commands to consume.
+		os.Setenv("INPUT", *inputDir)
+		os.Setenv("OUTPUT", *outputDir)
+
+		if *preprocessCommand != "" {
```

Comment:
The intent is to make it a built-in feature, so it's more clear to the users that it's a common practice to do pre-process/post-process when using `prowgen`. Otherwise they may consider adding non-generic feature to `prowgen`, or consider it hacky to do pre-process/post-process, neither of which will be ideal since `prowgen` itself cannot accommodate all use cases.